### ISSUE TITLE
Accounts Balance needs currency history to show data correctly

### ISSUE THREAD

**Issue by vomikan** [moneymanagerex/moneymanagerex#2456]
Subj.
-  Check MMEX version
  - [x] 1.3.x

#2437 
Regarding #2433 is better to reflect amount for today's currency rates.

**Comment by gabriele-v** [comment]
I think issue it's here: https://github.com/moneymanagerex/moneymanagerex/blob/778acf5ff4c3f47368b591591a01c4400e248d47/src/model/Model_CurrencyHistory.cpp#L95
I'm quite sure I've added a fix in 1.4.x branch that probably hasn't been backported, I'll have a look later

**Comment by vomikan** [comment]
It's more complex issue. Someone decided that historical data is fair for calculating the transaction costs of the past.
I think this is a fake. Everything must be presented at the current rate.

![image](https://user-images.githubusercontent.com/6836805/81916744-e83ee080-95dc-11ea-948b-194c30a1af49.png)

**Comment by gabriele-v** [comment]
Yep, it has been done by me. Why you are not agreeing with this?
Let's say that two weeks ago 1 EUR was 1,5 USD and I had choice to purchase the same object with the same price but in different currencies.
I've purchased it in USD, keep it simple I've paid 1,5 USD that was equal to 1 EUR.
Now at actual time 1 EUR it's 1 USD (exaggerated, just to keep example simple).
Report in base currency should show you that you have paid 1 EUR (so using currency history) or 0,66 (using actual rate)?
IMHO it should show you 1 EUR..

Problem is that Model_CurrencyHistory::getDayRate it's actually wrong, because if no currency histories are found, it uses 1 (that's the issue of other users are well), instead of using BASECONVRATE that should be a fixed rate used as "last resort" or in case user won't like to use CurrencyHistory as you.
Doing this we anyway let the user choice between the two methods (it should be documented obviously).

So my plan was to fix getDayRate and then revert your changes to HomePage (I think you have changed just that one, correct?)

**Comment by gabriele-v** [comment]
Maybe we could add a new settings "use currency history" that if enabled will use currency history as explained above by me, while if deactivated it always use baseconvrate and when you run update currencies, it just writes into baseconvrate the last rate, without doing anything to the history, what do you think?

It's easy to add because if that flag is disabled, simply there won't be any history and getDayRate will always return the baseconvrate. In this way we can have both behaviors at the same time

**Comment by vomikan** [comment]
Possible reason of misunderstanding is: In Russia forbidden purchase / sale  everything in foreign currencies. All sales and purchases are made in rubles.
Let's say I see a product and an online store for $ 1. The course (conditionally) is 50. I spent 50 rubles on the product or $ 1 in the past.
Today, the exchange rate is 100 rubles per dollar (the Saudi has reduced the price of oil).
I still spent $ 1 (virtually) and 50 rubles. (Not $.5 and 100RUB in current prices).

If I get a salary of 10,000 rubles, and spend it on various goods, including dollars, then buying at sales and when the rate is favorable does not mean that I have earned more.
In other words, I spend only rubles and it does not matter at what rate I convert them.

All this is difficult to explain in another language. I used google translator.

On other hand, you send me 100$ in the past (5000 RUB), Now it's 10000. True!
Becouse I did not converted it yet.

**Comment by vomikan** [comment]
> Today, the exchange rate is 100 rubles per dollar (the Saudi has reduced the price of oil).
I still spent $ 1 (virtually) and 50 rubles. (Not $.5 and 100RUB in current prices).

Seems your concept is correct.

IMHO better do not mix currencies. spent $ 1 and 50 rubles!

**Comment by vomikan** [comment]
Stop.
When I bot 1$ there was a transfer 1$ -> 50 RUB, then I spend 50RUB.
I did not spend 1$ to see it in Expeses report with current (100RUB) rate.

**Comment by vomikan** [comment]
I can’t decide for good.
But a simple solution, it seems to me better.
If you do not use the exchange rate, then it is easier and more understandable.

**Comment by gabriele-v** [comment]
I understood what you mean, but up to me your gain/loss due to exchange should be another transaction in another category called "exchange gain/loss" done when you transfer from USD account to RUB and vice-versa (than we could discuss it should be integrated into transfers but that's another point), because in your case the gain/loss was made on another date and on another time, not at the moment of the purchase.
Because if you change at the purchase, then you won't have any gain/loss (a part from bank rates but that's another point) and it will match with the history..

**Comment by gabriele-v** [comment]
Peronally considering it's already there and working, I won't delete everything, just put the checkbox.

I'm in doubt about the default, because it should be disabled by default (easier as you said) but if you already have some history inside because used in the past, it cannot be false. That's a matter where database upgrade it's really helpful, but don't know if we would like to release a new version or not.. Or there should be a function to get the default, but that's a trick due to poor DB side up to me.

**Comment by vomikan** [comment]
The original scheme without historical courses was simple and sufficient.
I transfer money (rubles) to the USD account at the rate of the store (not the central bank). In this translation, losses on commission and conversion are hidden, but I don’t care about it, because I really want to receive the goods.
We need to create a simple database and beat it. Virtually not very clear.

**Comment by vomikan** [comment]
I even thought that it is possible to simulate the exchange rate of a transfer transaction by simply calculating the average on the day of transfer.

**Comment by gabriele-v** [comment]
I not agree that without historical was sufficient.. Do not consider for a while the transfer issues, if I have an account in USD and one in EUR and with both I purchased electronics stuff, how can I understand how much I've spent overall (or for each year) in electronic stuffs? At the moment my MMEX database has a bit less than 10 years of transactions, how could I use a single rate for EUR/USD if (from real grahps) in 10 years it varied between 1.48 and 1.04?

**Comment by vomikan** [comment]
I already suggested not to mix currencies.
Then everything will be right.
If the user does not convert currencies, then this will be correct.

To understand the costs for today in the base (national) currency, it is enough to look at the exchange rate for today.

In this situation, you need to fix the bug, but so as not to complicate things too much.

**Comment by vomikan** [comment]
One more thing. I've adder ![image](https://user-images.githubusercontent.com/6836805/81930039-a9b32100-95f0-11ea-8f14-f90ef2574cb7.png) button to get online rates for currencies and stocks.
All values may be downloaded by one request. It's sinple and cool and fast.

To keep all used currencies and stock historical data it should be several queries to Yahoo.com with date ranges. It's would be difficult and slow. 

And what we would do in case if Yahoo close API ?
https://github.com/moneymanagerex/moneymanagerex/issues/1367

**Comment by gabriele-v** [comment]
This function will download only daily rates (and add to history if enabled).
If someone wants to download full history, it's possible to do it from main currency dialog.

Regarding Yahoo API it's another point and anyway both (history and fixed rate) will continue working if added manually
